src/seedu/addressbook/data/person/Address.java

```diff
@@ -27,15 +27,36 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        partitionAddress();
     }
-
     /**
      * Returns true if a given string is a valid person address.
      */
     public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
 
+    /**
+     *  Partitions address into its block, street, unit and postal code.
+     *
+     *  Assumes valid input.
+     */
+    private void partitionAddress() {
+        try {
+            String[] categories = value.split(",\\s", 4);
+        Block block = new Block(Integer.parseInt(categories[0]));
+        Street street = new Street(categories[1]);
+        Unit unit = new Unit(categories[2]);
+        PostalCode postalCode = new PostalCode(Integer.parseInt(categories[3]));
```

Comment:
Can remove magic numbers. Based on the UML diagram, the new variables should be declared as class attributes.

Reply:
Can I generalize that when I have a composition relationship, I must set the objects instantiated from the 'part' class as class attribute of the 'whole' class? (I am referring to the prof's term to treat composition as "whole-part" relationship)

Reply:
Yes.. Good..